pkg/registry/apis/provisioning/jobs/migrate/worker.go

```diff
@@ -84,17 +72,34 @@ func (w *MigrationWorker) Process(ctx context.Context, repo repository.Repositor
 		return errors.New("missing migrate settings")
 	}
 
+	progress.SetTotal(ctx, 10) // will show a progress bar
+	rw, ok := repo.(repository.ReaderWriter)
+	if !ok {
+		return errors.New("migration job submitted targeting repository that is not a ReaderWriter")
+	}
+	parser, err := w.parsers.GetParser(ctx, rw)
```

Comment:
moved here parser as it's common for both and we should fail fast before cloning